realsense_camera/src/base_nodelet.cpp

```diff
@@ -45,8 +45,12 @@ namespace realsense_camera
 
     stopCamera();
 
-    rs_delete_context(rs_context_, &rs_error_);
-    checkError();
+    if (rs_context_)
+    {
+      rs_delete_context(rs_context_, &rs_error_);
+      rs_context_ = NULL;
+      checkError();
+    }
 
     ROS_INFO_STREAM(nodelet_name_ << " - Stopping...");
     ros::shutdown();
```

Comment:
Can you please add a check ros::isShuttingDown() here so that we don't call the shutdown again?